clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/CompletableApplicationEvent.java

```diff
@@ -38,6 +43,13 @@ protected CompletableApplicationEvent(final Type type, final long deadlineMs) {
         this.deadlineMs = deadlineMs;
     }
 
+    protected CompletableApplicationEvent(final Type type, final long deadlineMs, final boolean isPassedByErrorEvent) {
+        super(type);
+        this.future = new CompletableFuture<>();
+        this.deadlineMs = deadlineMs;
+        this.isPassedByErrorEvent = isPassedByErrorEvent;
+    }
```

Comment:
I'd prefer to see the other constructor call this one so that we have as few paths as possible.